Sources/PackageGraph/DependencyResolver.swift

```diff
@@ -296,7 +334,12 @@ struct PackageContainerConstraintSet<C: PackageContainer>: Collection {
     ///
     /// - Returns: The new set, or nil the resulting set is unsatisfiable.
     func merging(_ constraint: PackageContainerConstraint<Identifier>) -> PackageContainerConstraintSet<C>? {
-        return merging(versionRequirement: constraint.versionRequirement, for: constraint.identifier)
+        switch constraint.requirement {
+        case .versionSet(let versionSet):
+            return merging(versionRequirement: versionSet, for: constraint.identifier)
+        case .unversioned:
+            return nil
```

Comment:
Why does this return nil? That seems wrong to me -- if I have a constraint set, merging in an unversioned requirement for a package should probably always replace any existing assignment for the package with an unversioned binding.

It can never cause a satisfiable constraint set to become unsatisfiable...